happi/tests/test_cli.py

```diff
@@ -60,19 +60,26 @@ def trim_split_output(strings: str, delim: str = '\n'):
 def assert_match_expected(
     result: click.testing.Result,
     expected_output: Iterable[str],
-    expected_return: int = 0
+    expected_return: int = 0,
+    raise_failures: bool = False
 ):
     """
     standard checks for a cli result, confirms output matches expected
     SystemExit exception is acceptable if the expected return code is some failure code
     """
     logger.debug(result.output)
     assert result.exit_code == expected_return
-    assert not result.exception or (isinstance(result.exception, SystemExit) and expected_return != 0)
+    assert not result.exception or (isinstance(result.exception, SystemExit)
+                                    and expected_return != 0)
 
     trimmed_output = trim_split_output(result.output)
     for message, expected in zip(trimmed_output, expected_output):
-        assert message == expected
+        if raise_failures:
+            assert message == expected
+        else:
+            if not message == expected:
+                print('Output does not match expected value:\n'
+                      f'- Output: ({message}\n- Expected: {expected}')
```

Comment:
I think the split here is a good call, it could be helpful debug information when the actual check fails